### DISCUSSION TITLE
Introduction and Discussion on the Upcoming Version 2 of PrestaSharp

### DISCUSSION THREAD

**Discussion by LameuleFR** [Bukimedia/PrestaSharp#467] [Announcements]
Hello PrestaSharp Community,

I am Benjamin, the new maintainer of the PrestaSharp project, and I'm excited to share my journey with you. I've been immersed in the world of e-commerce development using prestashop but currently spearheading a Blazor application initiative within my company. I've stepped into this role with enthusiasm and a commitment to bring new life into PrestaSharp after a three-year hiatus.

Here's what I have been working on for the upcoming Version 2:

- **\[MAJOR CHANGE\]** Almost all method are now async
- **\[CHANGE\]** PrestashopEntity base class is now partial
- **\[IMPROVEMENT\]** Upgrading our compatibility to .NET 8 to leverage the latest framework advancements and bring major performance gain.
- **\[IMPROVEMENT\]** Implementation of Multistore support with `id_shop` for enhanced multi-store management with PrestaShop.
- **\[IMPROVEMENT\]** Update to the latest RestSharp for improved stability and performance.
- **\[IMPROVEMENT\]** Adding HTTP/2 support, thank's to new RestSharp & HttpClient implementation.
- **\[IMPROVEMENT\]** Automatic caching to reuse HttpClient for multiple RestSharpFactory instance
- **\[IMPROVEMENT\]** Throw exception with explicit error when the XML response contain errors
- **\[IMPROVEMENT\]** Add GetAllParallelAsync() method to quickly retrieve large dataSet. (ex: 9sec to retrieve/deserialize 60k customer from a distant PS WebService)
- **\[IMPROVEMENT\]** Possibility to paginate result in getAll() method to prevent fail on large dataset
- **\[Fix\]** Intermediate certificate issue
- **\[Fix\]** Addressing connection pool starvation to optimize resource utilization.
- **\[Fix\]** Hanging open TCP connections
- **\[Fix\]** Stability issues

The development is in full swing, and while these improvements are shaping up, I'm actively working to ensure the library is working great. I am actively working on these features and continuously experimenting within my local branch to refine these updates before they are ready for release. The list is expected to evolve as development progresses and as we identify more areas for improvement.

Additionally, I'm planning to add a more efficient data retrieval system for large stores by enabling the `GetAllAsync()` method to fetch data in batches. This will ensure that even the largest stores can perform at optimal levels without overwhelming the server or encountering timeouts.

I'm eager to gather your insights on these initiatives. What features or fixes do you think are crucial for PrestaSharp's next iteration? Are there specific challenges you've encountered with the current version that you would like us to address?

Let's work together to make PrestaSharp a state-of-the-art tool for Prestashop platforms.

Benjamin

**Comment by totti240282** [comment]
Yeah!!!

**Comment by jarod-rossini** [comment]
I can't wait !

**Comment by Draneus** [comment]
Fantastic journey you're going for here! Solid base, wish you luck and success!

**Comment by LameuleFR** [comment]
**Last improvement from today:**

[IMPROVEMENT] Automatic caching of HttpClient to reuse it for multiple RestSharpFactory instance
[IMPROVEMENT] Throw exception with explicit error when the XML response contain error
[IMPROVEMENT] Possibility to paginate result in getAll() method to prevent fail on large dataset
[IMPROVEMENT] Add GetAllParallelAsync() method to quickly retrieve large dataset. (ex: 9sec to retrieve/deserialize 60k customer from a distant PS WebService)

**Comment by totti240282** [comment]
Any news ?

**Comment by marconline** [comment]
I would like to contribute to this project, but I see that the current version in the main branch is 1.1.0 while on Nuget there is version 1.2.9.
How can I do?

**Comment by LameuleFR** [comment]
Hello @totti240282  & @marconline,
I've tried to contact the repository owner to get access to the nugget package and straighten things out, but I've never gotten a reply.

For my part, I continued to develop the version for my internal needs. I'll publish it on another repository as a Fork very soon if you're interested.

**Reply by totti240282** [reply]
Yes I'm interested, new version is compatible with PS 8 ?

**Comment by marconline** [comment]
Yes, I'm interested. But did you start from this version?

**Reply by totti240282** [reply]
Yes I'm interested, new version is compatible with PS 8 ?
